### PR TITLE
overlay a click-interceptor for paypal when the amount chosen is too low

### PR DESCRIPTION
**Pull request by Pomax**
closes https://github.com/mozilla/donate-wagtail/issues/665

Note that for review purposes, the overlay is red because it has a solid background color set so you can see the overlay existing at all. Prior to landing, that background color will get removed, so no one will ever see it.

### CODE REVIEW

File: source/sass/abstracts/_variables.scss [hunk 1]
```diff
@@ -79,16 +79,17 @@ $breakpoints: (
 */
 
 $z-index: (
-  1: 10,
```
Comment (Pomax) [inline source/sass/abstracts/_variables.scss]: there is no scss that uses `@include zindex(1)`

File: source/sass/abstracts/_variables.scss [hunk 2]
```diff
@@ -79,16 +79,17 @@ $breakpoints: (
...
+  hidden: -1,
+  base: 1,
   to-do: 20,
   search: 30,
   nav: 40,
   header: 50,
   nav-desktop: 60,
   ui-overlay: 70,
   sticky-message: 80,
-  9: 90,
```
Comment (Pomax) [inline source/sass/abstracts/_variables.scss]: there is no scss that uses `@include zindex(9)`

File: source/sass/components/_payments.scss [hunk 3]
```diff
@@ -19,10 +19,22 @@
...
       }
+      &--overlay {
+        display: none;
+        position: absolute;
+        @include z-index(overlay);
+        top: 0;
+        right: 0;
+        bottom: 0;
+        left: 0;
+        background: $color--warning;
```
Comment (Pomax) [inline source/sass/components/_payments.scss]: this line will be removed once this is R+

File: webpack.config.js [hunk 4]
```diff
@@ -13,48 +13,25 @@ let rules = [
...
+  return {
+    entry: `./source/js/${name}.js`,
+    output: {
+      path: frontendPath,
+      filename: `${name}.compiled.js`
+    },
+    module: {
+      rules
+    },
+    devtool: `none` // see https://webpack.js.org/configuration/devtool/
```
Comment (Pomax) [inline webpack.config.js]: The `devtool` instruction was missing, which means it defaulted to creating a source map based on `eval`, which made _everything_ break.

Fuck magical implicit webpack nonsense, it absolutely destroys a day's worth of productivity.

**Comment by beccaklam** [comment]
Hey @Pomax so the changes are looking good but I have some questions:
1) What is the exact behaviour we're going for here? Should the comment box ("Please select a value ...") show up on every screen? It's working for me in Chrome for all screen widths, only desktop for Firefox (though I'm not sure if this is because of the Inspector, seems to work fine on Android on Browser Stack), and is very buggy is Safari. 
2) What is the red overlay you're referring to? IS it the red outline around the box?
Screen Shot 2020-01-13 at 3 52 29 PM

**Comment by Pomax** [comment]
@beccaklam you should see a red overlay on the paypal button when the custom amount is too low:

image

Also it turns out we don't control that message: it's default browser form validation behaviour, which on the one hand means it might not be a super nice error message, but on the other hand does mean that we get proper, vetted localization for free, so we probably don't want to fix that message in this PR (if at all).

**Comment by beccaklam** [comment]
> @beccaklam you should see a red overlay on the paypal button when the custom amount is too low:
> 
> image
> 
> Also it turns out we don't control that message: it's default browser form validation behaviour, which on the one hand means it might not be a super nice error message, but on the other hand does mean that we get proper, vetted localization for free, so we probably don't want to fix that message in this PR (if at all).

Thanks for the clarification, @Pomax. The functionality looks good, though I do agree with @alanmoo 's comment above 👆 And one design note: would it be possible to have the red outline go all the way around the box, including the '$' sign portion? It looks a bit odd cut off. If it's too big of a lift, that's okay we can still ship it.

Screen Shot 2020-01-14 at 10 29 30 AM

**Comment by alanmoo** [comment]
@beccaklam How's an inset error state instead?
image

```
#one-time-amount-other-input:invalid {
  box-shadow: red inset 0px 0px 1.5px 1px;
}
```

**Comment by beccaklam** [comment]
> @beccaklam How's an inset error state instead?
> image
> 
> ```
> #one-time-amount-other-input:invalid {
>   box-shadow: red inset 0px 0px 1.5px 1px;
> }
> ```

Oh that's much better. Thanks @alanmoo !

**Comment by Pomax** [comment]
The problem is mostly one of timing: we're listening for `input` and `change` events at the moment, because if we only listened to just `change` then the user can type a value and click the paypal button without the form every having "changed" first.

I'm not sure we can get around that without a lot more code and introducing some edge cases =(

**Comment by Pomax** [comment]
@alanmoo @beccaklam inset highlight added, but let's leave the timing issue as something separate.

**Comment by Pomax** [comment]
nice, then I'll remove the background color in preparation for landing this.

**Pull request merged**